<div class="chatlog__message-group">
  <div class="chatlog__author-avatar-container">
    <img class="chatlog__author-avatar" src="https://cdn.discordapp.com/avatars/127674083331538944/4893b43ea9ee2231b8ffb8b05c305ce3.png?size=128" alt="Avatar" loading="lazy">
  </div>
  <div class="chatlog__messages">
    <span class="chatlog__author-name" title="norio_nomura#8334" data-user-id="127674083331538944" style="color: rgb(155,89,182)">norio_nomura</span>
    <a href="/channels/481326746579369995?category=開発環境, ライブラリ&amp;channel=lldb&amp;message_id=535767987534168084"><span class="chatlog__timestamp">18-Jan-19 10:27 AM</span></a>
    <div class="chatlog__message " data-message-id="535767237827624970" id="message-535767237827624970" title="Message sent: 18-Jan-19 10:27 AM">
      <div class="chatlog__content">
        <div class="markdown">
          <span class="preserve-whitespace">Swift Error Breakpoint使ってlldbでエラーの詳細を見られないかと試したけど、ダメぽい。</span>
          <span class="chatlog__edited-timestamp" title="18-Jan-19 10:41 AM">(edited)</span>
        </div>
      </div>
    </div>
    <div class="chatlog__message " data-message-id="535767896350261258" id="message-535767896350261258" title="Message sent: 18-Jan-19 10:30 AM">
      <div class="chatlog__content">
        <div class="markdown">
          <span class="preserve-whitespace">SILを見ると<span class="pre pre--inline">willThrow</span>にErrorを渡してる。 <div class="pre pre--multiline nohighlight">// main sil @main : $@convention(c) (Int32, UnsafeMutablePointer&lt;Optional&lt;UnsafeMutablePointer&lt;Int8&gt;&gt;&gt;) -&gt; Int32 { bb0(%0 : $Int32, %1 : $UnsafeMutablePointer&lt;Optional&lt;UnsafeMutablePointer&lt;Int8&gt;&gt;&gt;): %2 = metatype $@thin MyError.Type // user: %4 // function_ref MyError.init() %3 = function_ref @main.MyError.init() -&gt; main.MyError : $@convention(method) (@thin MyError.Type) -&gt; @owned MyError // user: %4 %4 = apply %3(%2) : $@convention(method) (@thin MyError.Type) -&gt; @owned MyError // user: %7 %5 = alloc_existential_box $Error, $MyError // users: %10, %8, %6 %6 = project_existential_box $MyError in %5 : $Error // user: %7 store %4 to %6 : $*MyError // id: %7 %8 = builtin "willThrow"(%5 : $Error) : $() br bb1 // id: %9 bb1: // Preds: bb0 %10 = builtin "errorInMain"(%5 : $Error) : $() unreachable // id: %11 } // end sil function 'main'</div></span>
          <span class="chatlog__edited-timestamp" title="18-Jan-19 10:30 AM">(edited)</span>
        </div>
      </div>
    </div>
    <div class="chatlog__message " data-message-id="535767987534168084" id="message-535767987534168084" title="Message sent: 18-Jan-19 10:30 AM">
      <div class="chatlog__content">
        <div class="markdown">
          <span class="preserve-whitespace">けど、ブレークポイントで<span class="pre pre--inline">$arg2</span>を見ても<span class="pre pre--inline">0</span>になってる。 <div class="pre pre--multiline nohighlight">$ xcrun --toolchain org.swift.5020190117a swift swift_oss_helper command enabled. warning: (x86_64) /Library/Developer/Toolchains/swift-5.0-DEVELOPMENT-SNAPSHOT-2019-01-17-a.xctoolchain/System/Library/PrivateFrameworks/LLDB.framework/Resources/repl_swift empty dSYM file detected, dSYM was created with an executable with no debug info. Welcome to Apple Swift version 5.0-dev (LLVM a6b53a73a4, Clang 5fd77fa5bf, Swift 08e20c0d48). Type :help for assistance. 1&gt; :br s -E swift Breakpoint 1: where = libswiftCore.dylib`swift_willThrow at ErrorObject.mm:569, address = 0x0000000101313d50 1&gt; struct MyError: Error, CustomStringConvertible { 2. var description = "my error" 3. } 4. 5. throw MyError() Execution stopped at breakpoint. Enter LLDB commands to investigate (type help for assistance.) libswiftCore.dylib was compiled with optimization - stepping may behave oddly; variables may not be available. Process 68387 stopped * thread #1, queue = 'com.apple.main-thread', stop reason = breakpoint 1.1 frame #0: 0x0000000101313d50 libswiftCore.dylib`swift_willThrow(unused=0x0000000000000000, error=&lt;unavailable&gt;) at ErrorObject.mm:569 [opt] 566 /// Breakpoint hook for debuggers. 567 SWIFT_CC(swift) void 568 swift::swift_willThrow(SWIFT_CONTEXT void *unused, -&gt; 569 SWIFT_ERROR_RESULT SwiftError **error) { 570 // empty 571 } 572 Target 0: (repl_swift) stopped. (lldb) p $arg1 (unsigned long) $0 = 4296026880 (lldb) p $arg2 (unsigned long) $1 = 0 (lldb) </div>
            <s>(SILと同じソースで試したログと差し替え)</s> (シンボル入りtoolchainで試したログと差し替え)</span>
          <span class="chatlog__edited-timestamp" title="18-Jan-19 10:47 AM">(edited)</span>
        </div>
      </div>
    </div>
  </div>
</div>